### PR TITLE
Changes to support Lighttpd 1.4.49

### PR DESCRIPTION
**Pull request by dgarske**
* Fix for `wolfSSL_CTX_set_options` to work correctly when no certificate has been set for WOLFSSL_CTX, otherwise this operation fails with `Server missing certificate`.
* Fix for bad argument name `time`.
* Fix for `warning: type of bit-field`: Allowed types for bit-fields are int and unsigned int only.
* Exposed `ERR_remove_thread_state` and `SSL_CTX_set_tmp_ecdh` for lighttpd
* Renamed `WOLFSSL_ERR_remove_thread_state` to `wolfSSL_ERR_remove_thread_state` and setup old name macro.
* Add missing newline on asn1.h.
* Whitespace cleanup in ssl.c.

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -11328,15 +11328,16 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
 
     long wolfSSL_CTX_set_options(WOLFSSL_CTX* ctx, long opt)
     {
-        WOLFSSL *ssl;
+        WOLFSSL ssl;
```
Comment (toddouska) [inline src/ssl.c]: This puts a lot of memory on the stack.  Why the change from dynamic?

Comment (dgarske) [reply src/ssl.c]: @toddouska : The `wolfSSL_new()` function call was failing with `Server missing certificate` (https://github.com/wolfSSL/wolfssl/blob/master/src/internal.c#L4259) because the `ctx` did not yet have the server certificate loaded.

To resolve the increased stack I've pushed a fix to allocate the `ssl` from the heap.

File: src/ssl.c [hunk 2]
```diff
@@ -11330,15 +11330,21 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
...
+        WOLFSSL* ssl;
         WOLFSSL_ENTER("SSL_CTX_set_options");
-        if(ctx == NULL)
+        if (ctx == NULL)
             return BAD_FUNC_ARG;
-        ssl = wolfSSL_new(ctx);
-        if(ssl == NULL)
-            return SSL_FAILURE;
+
+        ssl = (WOLFSSL*)XMALLOC(sizeof(WOLFSSL), ctx->heap, DYNAMIC_TYPE_SSL);
```
Comment (toddouska) [inline src/ssl.c]: Why are we manually doing the malloc here vs using the _new APIs like before?

**Comment by toddouska** [comment]
David, maybe having Eric refactor set options to a use a mask from SSL or CTX is the best solution?

**Comment by dgarske** [comment]
@toddouska: I pushed a refactor to pull out the set options mask code. This resolves the issue with having to allocate a WOLFSSL object in `wolfSSL_CTX_set_options`.
